### PR TITLE
Added input arguments to use custom ort config files

### PR DESCRIPTION
**Pull request by sverhoeven**
## Description

This PR adds configuration files to the action, so we can configure rules, curations and classifications

Refs: #81

* * *

## Before you get started

- [x] I read and I followed [contributing guidelines](https://github.com/tortellini-tools/action/blob/main/CONTRIBUTING.md)
- [x] [☝ Create an issue](https://github.com/NLeSC/licenseguard/issues/new/choose) to discuss what you are going to do

## Checklist

This section should be filled in by the creator of this pull request to make sure the pull request is ready to review.

- [x] This pull request has a descriptive title
- [x] Code is written according to the code quality guidelines
- [x] Documentation if applicable is available
- [x] Tests run successfully
- [x] All checks below this pull request were successful

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: tortellini-tools/action@main
+      with:
+        curations: .tortelllin/curations.yml or https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/curations.yml
```
Comment (jspaaks) [inline README.md]:
```suggestion
        curations: .tortellli/curations.yml or https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/curations.yml
```

File: README.md [hunk 2]
```diff
@@ -32,6 +32,10 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: tortellini-tools/action@main
+      with:
+        curations: .tortellli/curations.yml or https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/curations.yml
```
Comment (jspaaks) [inline README.md]:
```suggestion
        curations: .tortellini/curations.yml or https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/curations.yml
```

**Comment by jspaaks** [comment]
I found some problems with this, e.g.

- node 12 doesnt understand the import style
- github action arguments that are required false should have a `default`
- github workflow didnt work without me adding a `return` in a couple places

Additions are in PR https://github.com/tortellini-tools/action/pull/109

File: README.md [hunk 3]
```diff
@@ -32,6 +32,10 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: tortellini-tools/action@main
+      with:
+        curations: .tortellini/curations.yml or https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/curations.yml
```
Comment (fdiblen) [inline README.md]: it is confusing to see `X or Y in` an example.
Can we have to separate examples:
1-using local config files
2-Fecthing config files from GitHub

File: action.yml [hunk 4]
```diff
@@ -4,8 +4,19 @@ author: 'Netherlands eScience Center'
 inputs:
   repositories:
     required: false
-    description: 'A file contains list of GitHub repository urls'
+    description: 'A file containing list of GitHub repository urls'
     default: ''
+  curations:
+    required: false
+    description: 'A file or URL containing curations correct invalid or missing package metadata and set the concluded license for packages. See https://github.com/oss-review-toolkit/ort/blob/master/docs/config-file-curations-yml.md'
```
Comment (fdiblen) [inline action.yml:11]: default value for curations is missing.

Comment (fdiblen) [inline action.yml:11]:
```suggestion
    description: 'A file or URL containing curations correct invalid or missing package metadata and set the concluded license for packages. See https://github.com/oss-review-toolkit/ort/blob/master/docs/config-file-curations-yml.md'
    default: https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/curations.yml
```

File: src/action.ts [hunk 5]
```diff
@@ -1,9 +1,11 @@
 import * as core from '@actions/core'
 import {check_directory, check_urls} from './check'
+import {set_up_configuration} from './config'
 
 export async function main(): Promise<void> {
     try {
         const repositories: string = core.getInput('repositories')
```
Comment (fdiblen) [inline src/action.ts:7]: what about other inputs like curations, rules, classifications?
I think you should also use `getInput` for these variables.

File: src/action.ts [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
+        await set_up_configuration()
```
Comment (fdiblen) [inline src/action.ts:8]: Once you have configuration related variables, you need to use them as an argument in set_up_configuration
like
```typescript
await set_up_configuration(curations, rules,classifications)
```

**Review by fdiblen — CHANGES_REQUESTED** [review]
Well done!
I left a minor comments. Can you please fix the test-action and improve the PR description?

File: action.yml (hunk 4 above)
Comment (jspaaks) [reply action.yml:11]: fixed in later PR, see https://github.com/tortellini-tools/action/pull/109#discussion_r644211308

Comment (jspaaks) [inline action.yml:11]: fixed in https://github.com/tortellini-tools/action/pull/109#discussion_r644211308, we chose to not have default curations so the arg is initialized as empty string

File: README.md (hunk 3 above)
Comment (jspaaks) [reply README.md]: I chose to just remove getting the curations file from a URL, as this is not typical behavior. curations are expected to live next to the code in the repo

**Review by fdiblen — APPROVED** [review]

**Comment by jspaaks** [comment]
Thanks for the review

**Pull request merged**